lib/proxy/util.rb

```diff
@@ -11,16 +11,21 @@ class CommandTask
     # create new task and spawn new thread logging all the cmd
     # output to the proxy log. only the process' output is connected
     # stderr is redirected to proxy error log, stdout to proxy debug log
-    def initialize(acommand)
-      @command = acommand
+    # command can be either string or array (command + arguments)
+    # input is passed into STDIN and must be string
+    def initialize(command, input = nil)
+      @command = command
+      @input = input
     end
 
     def start(&ensured_block)
       # run the task in its own thread
-      logger.debug "Starting task: #{@command}"
-      @task = Thread.new(@command) do |cmd|
+      @task = Thread.new(@command, @input) do |cmd, input|
         status = nil
-        Open3.popen3(cmd) do |stdin, stdout, stderr, thr|
+        Open3.popen3(*cmd) do |stdin, stdout, stderr, thr|
```

Comment:
Coming from Python, it surprised me you can do this with strings. In Python `my-cmd` would end up as `['m', 'y', '-', 'c', 'm', 'd']`. I learned that here Python and Ruby differ.

Reply:
Yeah I tested this. There's a test for this as well.

Reply:
I did indeed test it in a shell. Just one of those Today I Learned moments.